lib/rex/post/meterpreter/extensions/stdapi/sys/registry.rb

```diff
@@ -267,10 +281,17 @@ def Registry.query_value_direct(root_key, base_key, name, perm = KEY_READ)
     type = response.get_tlv(TLV_TYPE_VALUE_TYPE).value
     data = response.get_tlv(TLV_TYPE_VALUE_DATA).value
 
-    if type == REG_SZ
+    case type
+    when REG_DWORD
+      data = data.unpack1('L>')
+    when REG_EXPAND_SZ
+      data = data[0..-2]
+    when REG_MULTI_SZ
+      data = data[0..-3].split("\x00".b)
+    when REG_QWORD
+      data = data.unpack1('Q<')
+    when REG_SZ
       data = data[0..-2]
-    elsif type == REG_DWORD
-      data = data.unpack('N')[0]
     end
```

Comment:
Similar to other comment this and `query_value` look like common logic a private helper may consolidate.